### DISCUSSION TITLE
As a Debug/Trace Device

### DISCUSSION THREAD

**Discussion by pfiliberti** [markadev/AppleII-VGA#23] [Ideas]
I was thinking of how this could be used to debug an Apple Motherboard I'm having issues with. It would be interesting if all it did was log the bus activity, ADDRESS R/W Maybe decode the instruction and keep in a log that could be scrolled for so many instruction cycles. It could be used to see if things are working from fetching rom instructions on power up to if r/w to ram is functional etc..... Just a thought..

**Comment by jtflanagan** [comment]
[My AppleTini project](https://github.com/jtflanagan/AppleTini/) isn't fully functional yet, but is pretty much exactly what you're asking about (and a bit more).  I did a breadboard demo of it at Kansasfest 2023 this year, working on a PCB version now.  The original desire was something to replace the VidHD to provide HDMI-out, but the way we are choosing to accomplish it is to literally the entire Apple II bus out over 100Mbit Ethernet, where a Raspberry Pi (or even just a desktop) listening on a socket can do whatever it wants with the bus data.  Video display is a trivial application of course.  But the whole bus is there so it could also have plenty of other applications, including the debugging you're looking for.
